### PR TITLE
Create interfaces UserDao, ShopCartDao and OrderDao with appropriate …

### PR DESCRIPTION
**Pull request by fyarkiy**
…...Impl classes, Create services (interfaces and Classes) for Orders, ShopCart and Users. Add execution lines in Application class to check the build

### CODE REVIEW

File: src/main/java/internet/store/dao/impl/ShopCartDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public Optional<ShoppingCart> getByUserId(Long userID) {
+        return Storage.shopCarts.stream()
+                .filter(s -> s.getUserId().equals(userID))
+                .findFirst();
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        IntStream.range(0, Storage.shopCarts.size())
+                .filter(p -> shoppingCart.getCartId().equals(Storage.shopCarts.get(p).getCartId()))
```
Comment (Borisko130) [inline src/main/java/internet/store/dao/impl/ShopCartDaoImpl.java]: Line is longer than 100 characters

File: src/main/java/internet/store/Application.java [hunk 2]
```diff
@@ -12,10 +17,11 @@
     public static void main(String[] args) {
         ProductService productService = (ProductService) injector.getInstance(ProductService.class);
```
Comment (Borisko130) [inline src/main/java/internet/store/Application.java:18]: Probably it is better to make service classes instances (productService, userService etc.) class variables.

File: src/main/java/internet/store/model/ShoppingCart.java [hunk 3]
```diff
@@ -36,4 +36,16 @@ public void setProducts(List<Product> products) {
     public void setUserId(Long userId) {
         this.userId = userId;
     }
+
+    @Override
+    public String toString() {
+        return "ShoppingCart{"
```
Comment (Borisko130) [inline src/main/java/internet/store/model/ShoppingCart.java]:
```suggestion
        return "ShoppingCart{ cartId=" cartId
         + ", products=" + products
         + ", userId=" + userId + '}';
```

File: src/main/java/internet/store/model/User.java [hunk 4]
```diff
@@ -43,4 +43,16 @@ public String getPassword() {
     public void setPassword(String password) {
         this.password = password;
     }
+
+    @Override
+    public String toString() {
+        return "User{"
```
Comment (Borisko130) [inline src/main/java/internet/store/model/User.java]:
```suggestion
        return "User{ userName=" + userName
                + ", logIn=" + logIn  + '}';
```

File: src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shopCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().remove(product);
+        shopCartDao.update(shoppingCart);
+        return !shoppingCart.getProducts().removeIf(p -> p.equals(product));
```
Comment (Borisko130) [inline src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java]: I don't understand what this does

File: src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shopCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().remove(product);
```
Comment (Borisko130) [inline src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        shopCartDao.update(shoppingCart).getProducts().remove(product);
```

**Review by Borisko130 — APPROVED** [review]

File: src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java (hunk 5 above)
Comment (fyarkiy) [reply src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java]: Return "true" if there are no such product anymore in the Cart (i.e. operation in line 29 was successful and attempt to remove this product second time is failed)

File: src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java (hunk 6 above)
Comment (fyarkiy) [reply src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java]: Doesn't work this way as remove(product) return boolean and I can't use shopCartDao.update. can be done throw remove(product id) but it seems to me it'll be longer

File: src/main/java/internet/store/Application.java (hunk 2 above)
Comment (fyarkiy) [reply src/main/java/internet/store/Application.java:18]: Did you mean - final? might be  but not sure...

File: src/main/java/internet/store/dao/impl/ShopCartDaoImpl.java (hunk 1 above)
Comment (fyarkiy) [reply src/main/java/internet/store/dao/impl/ShopCartDaoImpl.java]: No comments from maven but I'll split for better readability

File: src/main/java/internet/store/Application.java (hunk 2 above)
Comment (Borisko130) [reply src/main/java/internet/store/Application.java:18]: I mean move them outside of main method but I'm not sure as well)

File: src/main/java/internet/store/dao/impl/OrderDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        IntStream.range(0, Storage.orders.size())
+                .filter(o -> Storage.orders.get(o).getOrderId().equals(order.getOrderId()))
+                .forEach(i -> Storage.orders.set(i, order));
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.orders
+                .removeIf(o -> o.getOrderId().equals(id));
```
Comment (Sofasmile) [inline src/main/java/internet/store/dao/impl/OrderDaoImpl.java]:
```suggestion
        return Storage.orders.removeIf(o -> o.getOrderId().equals(id));
```

File: src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java (hunk 5 above)
Comment (Sofasmile) [inline src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        boolean result = shoppingCart.getProducts().remove(product);
        shopCartDao.update(shoppingCart);
        return result;
```

File: src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getProducts().clear();
+        shopCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUserId(Long userID) {
+        return shopCartDao.getByUserId(userID).get();
```
Comment (Sofasmile) [inline src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java]:
```suggestion
        return shopCartDao.getByUserId(userId).get();
```

File: src/main/java/internet/store/service/ShoppingCartService.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    ShoppingCart addProduct(ShoppingCart shoppingCart, Product product);
+
+    boolean deleteProduct(ShoppingCart shoppingCart, Product product);
+
+    void clear(ShoppingCart shoppingCart);
+
+    ShoppingCart getByUserId(Long userID);
```
Comment (Sofasmile) [inline src/main/java/internet/store/service/ShoppingCartService.java]:
```suggestion
    ShoppingCart getByUserId(Long userId);
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**